specification/dartLangSpec.tex

```diff
@@ -6794,12 +6796,14 @@ \subsubsection{String Interpolation}
 `\code{$s_0$\$\{$e_1$\}$s_1\ldots{}s_{n-1}$\$\{$e_n$\}$s_{n}$}' (where any of $s_0, \ldots, s_n$ can be empty)
 is evaluated by evaluating each expression $e_i$ ($1 \le i \le n$) in to a string $r_i$ in the order they occur in the source text, as follows:
 \begin{itemize}
-\item{} Evaluate $e_i$ to an object $o_i$.
-\item{} Invoke the \code{toString} method on $o_i$ with no arguments, and let
- $r_i$ be the returned value.
-\item{} If $r_i$ is not an instance of the built-in type \code{String}, throw an \code{Error}.
+\item Evaluate $e_i$ to an object $o_i$.
+\item Invoke the \code{toString} method on $o_i$ with no arguments,
+  and let $r_i$ be the returned object.
+\item If $r_i$ is not an instance of the built-in type \code{String},
+  throw an \code{Error}.
```

Comment:
In Dart 2, that has to mean that the returned value is `null`, because the return type of `Object.toString` is `String`.
(In Dart 2.x, the return type will be non-nullable `String` and this can be removed completely).

Reply:
Adjusted to say `If $r_i$ is the null object, a dynamic error occurs.`